src/PrestaShopBundle/Service/DataProvider/Marketplace/ApiClient.php

```diff
@@ -362,11 +371,42 @@ public function setPassword($password)
         return $this;
     }
 
+    public function setTimeout(int $timeout): self
+    {
+        $this->queryParameters['timeout'] = (string) $timeout;
+        $this->addonsApiClient->setDefaultOption('timeout', $timeout);
```

Comment:
@PierreRambaud we can use `setDefaultOption` here because the default value is reset by the `reset` method (just like the query parameters) Which is called before each use of this client in the `AddonsDataProvider` service